package.json

```diff
@@ -107,19 +140,24 @@
 		"test": "npm run compile && node ./node_modules/vscode/bin/test"
 	},
 	"devDependencies": {
-		"typescript": "^2.6.1",
-		"vscode": "^1.1.6",
-		"tslint": "^5.8.0",
 		"@types/chai": "^4.1.7",
-		"@types/node": "^7.0.43",
 		"@types/mocha": "^2.2.42",
-		"@types/sinon": "^7.0.11",
-		"@types/sinon-chai": "^3.2.2"
+		"@types/node": "^7.0.43",
+		"@types/sinon": "^7.0.13",
+		"@types/sinon-chai": "^3.2.2",
+		"nock": "^10.0.6",
+		"tslint": "^5.18.0",
+		"typescript": "^2.6.1",
+		"vscode": "^1.1.34"
 	},
 	"dependencies": {
+		"@types/nock": "^10.0.3",
+		"@types/request-promise": "^4.1.44",
 		"chai": "^4.2.0",
 		"child_process": "^1.0.2",
 		"path": "^0.12.7",
+		"request": "^2.88.0",
+		"request-promise": "^4.2.4",
```

Comment:
based on Paolo comment on using fetch instead of request-promise would worth looking at it but I think it is fine if we are looking at it in a second iteration (also fine if you modify it directly in this PR, it will just delay it a bit)

Reply:
I got things working with curl, but will definitely look into the fetch approach to replace request-promise after we get things mostly nailed down. I'd rather do that in a separate PR.